protoc-gen-combo/combo.go

```diff
@@ -66,10 +68,17 @@ func (this MixMatch) Gen(folder string, news []string) {
 			panic(fmt.Errorf("found another string {%s} after it was replaced with {%s}", old, news[i]))
 		}
 	}
-	if err = ioutil.WriteFile(filepath.Join(folder, this.Filename), []byte(content), 0666); err != nil {
+
+	// Replace package name
+	re := regexp.MustCompile(`\npackage (.*);`)
+	correctPkgName := strings.Trim(strings.ReplaceAll(folder, "/", "."), ".")
+	replaceStr := fmt.Sprintf("\npackage $1.%s;", correctPkgName)
+	content = re.ReplaceAllString(content, replaceStr)
```

Comment:
Non-trivial hack/fix.

`protoc-gen-combo` generates, from a single proto file, 3 or 4 other derived proto files in subfolders, with minimal changes that are defined below in this file. It simply copies the original file, and modified the pasted file using simple string search and replace.

I added this piece of logic here so that the derived proto file has correct package name.